WebService/ComparisonEngineUI/ComparisonEngineUI/Views/AddBarPage.xaml.cs

```diff
@@ -5,15 +5,63 @@
 using System.Threading.Tasks;
 using Xamarin.Forms;
 using Xamarin.Forms.Xaml;
+using ComparisonEngineUI.Data;
+using ComparisonEngineUI.Models;
+using ComparisonEngineUI.ViewModels;
 
 namespace ComparisonEngineUI.Views
 {
     [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class AddBarPage : ContentPage
     {
+        private List<Bar> _barList { get; set; }
+        public List<Bar> BarList
+        {
+            get
+            {
+                return _barList;
+            }
+            set
+            {
+                if (value != _barList)
+                {
+                    _barList = value;
+                    OnPropertyChanged();
+                }
+            }
+        }
         public AddBarPage()
         {
             InitializeComponent();
+            this.BindingContext = new AddBarViewModel();
+        }
+
+        private async void OnSaveClicked(object sender, EventArgs e)
+        {
+
+            if (string.IsNullOrEmpty(barName.Text) || string.IsNullOrEmpty(barAdress.Text))
+            {
+                await DisplayAlert("Warning", "Bad Input", "Ok");
+                return;
+            }
+            
+            Bar bar = new Bar(barName.Text,barAdress.Text);
+
+            var restService = new RestService();
+            BarList = Task.Run(async () => await restService.GetData<List<Bar>>(Constants.BarsUrl)).Result;
+
+            foreach (Bar searchBar in BarList)
+            {
+                
+                if (searchBar.BarName == bar.BarName && searchBar.BarLocation == bar.BarLocation)
```

Comment:
searchBar might conflict later on with an actual searchbar

Reply:
I agree. Actually, a lot of names given should be renamed for better readability of the code, will do that later.